### ISSUE TITLE
path details with 'time' returns server error

### ISSUE THREAD

**Issue by karussell** [graphhopper/graphhopper#1216]
Example query is [this](https://graphhopper.com/maps/?point=46.796534%2C-71.277752&point=46.820737%2C-71.300411&point=46.807345%2C-71.302643&details=time) and if you move the second point (the only via point) then it starts to get valid.

```
2017-11-09 20:12:27,793 [qtp1373419525-173] WARN  org.eclipse.jetty.server.HttpChannel - /route
java.lang.IllegalStateException: PathDetail list time is inconsistent due to entries 61566 [237, 244] vs. 12510 [245, 249]
        at com.graphhopper.util.PathSimplification.simplify(PathSimplification.java:137)
        at com.graphhopper.util.PathMerger.doWork(PathMerger.java:144)
        at com.graphhopper.routing.template.ViaRoutingTemplate.isReady(ViaRoutingTemplate.java:155)
        at com.graphhopper.GraphHopper.calcPaths(GraphHopper.java:1072)
```

I.e. 245 is one too much or 244 is one too small.

**Comment by karussell** [comment]
The problem is likely in PathMerger and it should be:

altRsp.addPathDetails(path.calcDetails(requestedPathDetails, pathBuilderFactory, origPoints == 0 ? 0 : origPoints - 1));

I think we currently only have A-B-C where at B we do a U-turn, but we also need a "straight" case where this error happened (I think - a bit too late now ;))

**Comment by karussell** [comment]
Hmmh, when we change this the test GraphHopperIT.testMonacoPathDetails fails at:

assertEquals(arsp.getPoints().size() - 1, detailList.get(0).getLast());

I think this is because the points of both paths are added (the via point is duplicate) but for the path details we do the correct `lastIndex += edge.fetchWayGeometry(2).size();` instead of `lastIndex += edge.fetchWayGeometry(3).size();` in PathDetailsFromEdges. @boldtrn do you think this could be the reason?

**Comment by boldtrn** [comment]
Yes, I think the reason is similar to the issue reported in #1138. I think we lingered too long in #1138, because removing the points from the Instructions was not that easy. I would propose that we fix this issue now by remove the duplicated point for the ViaInstructions.

> Hmmh, when we change this then the test GraphHopperIT.testMonacoPathDetails fails at

Yes this is expected I think :). I will create a fresh PR for this issue.

**Comment by karussell** [comment]
This is fixed via #1219 thanks @boldtrn !
